tools/cldr-apps/src/main/java/org/unicode/cldr/web/STFactory.java

```diff
@@ -230,11 +230,22 @@ private VoteResolver<String> setValueFromResolver(String path, VoteResolver<Stri
             }
             delegate.removeValueAtDPath(path);
             if (value != null) {
+                value = reviseInheritanceAsNeeded(path, value);
                 delegate.putValueAtPath(fullPath, value);
             }
             return resolver;
         }
 
+        public String reviseInheritanceAsNeeded(String path, String value) {
+            if (value != null) {
+                CLDRFile cldrFile = make(ballotBox.locale.getBaseName());
```

Comment:
I know this is closed ,but yes, this will return a resolving CLDRFile for the specific locale (the result is cached).